### PR TITLE
Ensure only a single connection is created between each server

### PR DESCRIPTION
**Pull request by kuujo**
This PR fixes a bug where multiple connections could be in progress between servers. It simply tracks connection futures in addition to the connections themselves.

### CODE REVIEW

File: server/src/main/java/io/atomix/copycat/server/state/ConnectionManager.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
 public final class ConnectionManager {
   private final Client client;
   private final Map<Address, Connection> connections = new HashMap<>();
+  private final Map<Address, CompletableFuture<Connection>> futures = new HashMap<>();
```
Comment (jhall11) [inline server/src/main/java/io/atomix/copycat/server/state/ConnectionManager.java]: Just my personal preference, so take it or leave it:
Maybe change futures to something more descriptive like "connectionFutures"?

**Comment by kuujo** [comment]
Restarting tests

File: server/src/main/java/io/atomix/copycat/server/state/ConnectionManager.java (hunk 1 above)
Comment (kuujo) [inline server/src/main/java/io/atomix/copycat/server/state/ConnectionManager.java]: I concur. Fixed.

**Comment by kuujo** [comment]
This seems to be the culprit causing Copycat tests to fail so frequently on forks:
https://github.com/travis-ci/travis-ci/issues/3775

**Pull request merged**